From 24b8cb7b2ac05616c9986a6408ab06911382f0c7 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sun, 10 Nov 2013 15:05:56 -0500
Subject: [PATCH] Fix SD card mounting

Based on Team-Hydra's commits:
- 6bcec8ceeaaa2a679aaa8948a67b9efdad06ad9f
- 884e730af2ce3f892078553763d885d770e88429

Change-Id: Id50baffaed87dda2c065587b3ee7c6eb68e40a91
---
 rootdir/etc/fstab.qcom   | 4 ++--
 rootdir/etc/init.qcom.rc | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/rootdir/etc/fstab.qcom b/rootdir/etc/fstab.qcom
index fed1c96..3faf6f0 100644
--- a/rootdir/etc/fstab.qcom
+++ b/rootdir/etc/fstab.qcom
@@ -16,5 +16,5 @@
 /dev/block/platform/msm_sdcc.1/by-name/efs            /efs          ext4    nosuid,nodev,noatime                        wait,check
 /dev/block/platform/msm_sdcc.1/by-name/userdata       /data        ext4    nosuid,nodev,noatime,noauto_da_alloc,journal_async_commit,errors=panic      wait,check,encryptable=footer,length=-16384
 
-/devices/platform/msm_sdcc.2/mmc_host/mmc2			/storage/sdcard1 		auto	default		voldmanaged=sdcard:auto
-/devices/platform/msm_hsusb_host/usb				/storage/usbdisk0		auto	default		voldmanaged=sda:auto
+/devices/platform/msm_sdcc.2/mmc_host/mmc2			auto 		auto	default		voldmanaged=sdcard1:auto
+/devices/platform/msm_hsusb_host/usb				auto		auto	default		voldmanaged=sda:auto
diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index ad972de..4877d5a 100644
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -623,7 +623,7 @@ service profiler_daemon /system/bin/profiler_daemon
     disabled
 
 # virtual sdcard daemon running as media_rw (1023)
-service sdcard /system/bin/sdcard /data/media /mnt/shell/emulated 1023 1023
+service sdcard /system/bin/sdcard -u 1023 -g 1023 -l /data/media /mnt/shell/emulated
     class late_start
     oneshot
 
-- 
1.8.4.2

